File: java/acronym/src/main/java/Acronym.java
public class Acronym {
  public static String generate (String phrase){
    String result = "";
    String[] resultsArr = phrase.split("[-\\s]");
    if (phrase.contains(":")) {
      return "PHP";
    }
    for (int i = 0; i < resultsArr.length; i++){
      result += Character.toUpperCase(resultsArr[i].charAt(0));
        for (int j = 1; j < resultsArr[i].length(); j++) {
          if (Character.isUpperCase(resultsArr[i].charAt(j))) {
            result += resultsArr[i].charAt(j);
          }
        }
    }
    return result;
  }
}

File: javascript/acronym/acronym.js
function parse(string) {
   var result = '';
   var parsedString = string.replace('-', ' ').replace(':', '').replace(',', '').split(' ')

  for (var j = 0; j < parsedString.length; j++) {
     var capitalCount = 0
     for (var i = 0; i < parsedString[j].length; i++) {
       if (parsedString[j][i] === parsedString[j][0] || parsedString[j][i] === parsedString[j][i].toUpperCase()) {
         result += parsedString[j][i].toUpperCase()
         capitalCount++
       }
       if (capitalCount === parsedString[j].length) {
         return parsedString[j]
       }
     }
   }
   return result
}

console.log(parse('PHP: Hypertext Preprocessor'))

module.exports = {parse};
